dtool/src/dtoolbase/dtoolbase.h

```diff
@@ -152,11 +152,11 @@ typedef struct _typeobject PyTypeObject;
 #ifdef PHAVE_TYPES_H
 #include <types.h>
 #endif // PHAVE_TYPES_H
-
+#if !defined(__wasi__)
 #ifdef PHAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif // PHAVE_SYS_TYPES_H
```

Comment:
Build system should simply not set PHAVE_SYS_TYPES_H for wasi if it doesn't have it, that's what these defines are for

Reply:
indeed, got that one fixed my cross toolchain file was not robust enough and was peeking into host /usr